TechAndCheck/zenodotus
# Update Rubocop to Ruby 3.1.2

This updates Rubocopp so it's ready for the newest version of Ruby. In the process of that it also bumps Ruby from 3.1.1 to 3.1.2.

Here we also set the new JS-style hash syntax to cause an error in Rubocop. Personally I'm just not a fan of it, it makes refactoring more difficult and you have to track down variables left and right when debugging.

After all this we run `rubocop -A` which fixes a number of auto-correctable style issues.

To test:
- Make sure you're running Ruby 3.1.2 `ruby -v`
- `bundle install`
- `rubocop` to check everything
- `rails t` to test

Thanks to @reefdog for bringing this to my attention.
